year2024/day3/Day3Step1.java

```diff
@@ -0,0 +1,165 @@
+package advent_of_code.day3;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Day3Step1 {
+    private static final String PART1_SAMPLE_FILE_NAME = ".\\src\\advent_of_code\\day3\\part1_sample.txt";
+    private static final String PART2_SAMPLE_FILE_NAME = ".\\src\\advent_of_code\\day3\\part2_sample.txt";
+    private static final String INPUT_FILE_NAME = ".\\src\\advent_of_code\\day3\\input.txt";
+    
+    private static final String INSTRUCTION_MUL_STR = "mul";
+    private static final String INSTRUCTION_DONT = "don't()";

```

Comment:
do() don't() に関する説明が https://adventofcode.com/2024/day/3 で見つけられませんでした。これらに関する仕様はどこに書かれていますでしょうか？

Reply:
Part1,Part2の二部構成でPart1を解かないとPart2の問題が見れないようになっています

Reply:
@nodchip さん @hayashi-ay さん
ありがとうございます。
すみません、Part2 の問題文を載せておりませんでした。
以後、載せるようにします。
この問題の Part2 については、この後のコメントで掲載いたしますね。

Reply:
--- Part Two ---

As you scan through the corrupted memory, you notice that some of the conditional statements are also still intact. If you handle some of the uncorrupted conditional statements in the program, you might be able to get an even more accurate result.

There are two new instructions you'll need to handle:

    The do() instruction enables future mul instructions.
    The don't() instruction disables future mul instructions.

Only the most recent do() or don't() instruction applies. At the beginning of the program, mul instructions are enabled.

For example:

xmul(2,4)&mul[3,7]!^don't()_mul(5,5)+mul(32,64](mul(11,8)undo()?mul(8,5))

This corrupted memory is similar to the example from before, but this time the mul(5,5) and mul(11,8) instructions are disabled because there is a don't() instruction before them. The other mul instructions function normally, including the one at the end that gets re-enabled by a do() instruction.

This time, the sum of the results is 48 (2*4 + 8*5).

Handle the new instructions; what do you get if you add up all of the results of just the enabled multiplications?